### ISSUE TITLE
java.lang.NullPointerException , if Progressview add android:visibility="gone"

### ISSUE THREAD

**Issue by abiang** [rey5137/material#91]
Hi! This library is very nice, Thank you.
My English is very bad, I hope you know what I say ......

if i delete the xml (android:visibility="gone" ) , app is work for Normal.

xml:    android:visibility="gone"
```javascript

<com.rey.material.widget.ProgressView
            android:id="@+id/progress"
            android:layout_width="18dp"
            android:layout_height="18dp"
            android:visibility="gone"
            app:pv_autostart="true"
            app:pv_circular="true"
            app:pv_progressStyle="@style/Material.Drawable.CircularProgress"
            app:pv_progressMode="indeterminate"/>
```

logcat:
```javascript
 Caused by: java.lang.NullPointerException
            at com.rey.material.widget.ProgressView.getProgressMode(ProgressView.java:125)
            at com.rey.material.widget.ProgressView.onVisibilityChanged(ProgressView.java:98)
            at android.view.View.dispatchVisibilityChanged(View.java:7499)
            at android.view.View.setFlags(View.java:8501)
            at android.view.View.<init>(View.java:3695)
            at com.rey.material.widget.ProgressView.<init>(ProgressView.java:45)
            at com.rey.material.widget.ProgressView.<init>(ProgressView.java:35)
```
all:
```javascript
05-10 14:16:27.534  27600-27600/tv.littlemuisc.android E/AndroidRuntime﹕ FATAL EXCEPTION: main
    android.view.InflateException: Binary XML file line #25: Error inflating class com.rey.material.widget.ProgressView
            at android.view.LayoutInflater.createView(LayoutInflater.java:613)
            at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:687)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:746)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
            at tv.littlemuisc.android.ui.adapter.HomeTestAdapter.onCreateViewHolder(HomeTestAdapter.java:93)
            at android.support.v7.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:4121)
            at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:3431)
            at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:3340)
            at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:1810)
            at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1306)
            at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1269)
            at android.support.v7.widget.LinearLayoutManager.scrollBy(LinearLayoutManager.java:1097)
            at android.support.v7.widget.LinearLayoutManager.scrollVerticallyBy(LinearLayoutManager.java:957)
            at android.support.v7.widget.RecyclerView$ViewFlinger.run(RecyclerView.java:2797)
            at android.view.Choreographer$CallbackRecord.run(Choreographer.java:749)
            at android.view.Choreographer.doCallbacks(Choreographer.java:562)
            at android.view.Choreographer.doFrame(Choreographer.java:531)
            at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:735)
            at android.os.Handler.handleCallback(Handler.java:725)
            at android.os.Handler.dispatchMessage(Handler.java:92)
            at android.os.Looper.loop(Looper.java:137)
            at android.app.ActivityThread.main(ActivityThread.java:5083)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:511)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560)
            at dalvik.system.NativeStart.main(Native Method)
     Caused by: java.lang.reflect.InvocationTargetException
            at java.lang.reflect.Constructor.constructNative(Native Method)
            at java.lang.reflect.Constructor.newInstance(Constructor.java:417)
            at android.view.LayoutInflater.createView(LayoutInflater.java:587)
            at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:687)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:746)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
            at tv.littlemuisc.android.ui.adapter.HomeTestAdapter.onCreateViewHolder(HomeTestAdapter.java:93)
            at android.support.v7.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:4121)
            at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:3431)
            at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:3340)
            at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:1810)
            at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1306)
            at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1269)
            at android.support.v7.widget.LinearLayoutManager.scrollBy(LinearLayoutManager.java:1097)
            at android.support.v7.widget.LinearLayoutManager.scrollVerticallyBy(LinearLayoutManager.java:957)
            at android.support.v7.widget.RecyclerView$ViewFlinger.run(RecyclerView.java:2797)
            at android.view.Choreographer$CallbackRecord.run(Choreographer.java:749)
            at android.view.Choreographer.doCallbacks(Choreographer.java:562)
            at android.view.Choreographer.doFrame(Choreographer.java:531)
            at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:735)
            at android.os.Handler.handleCallback(Handler.java:725)
            at android.os.Handler.dispatchMessage(Handler.java:92)
            at android.os.Looper.loop(Looper.java:137)
            at android.app.ActivityThread.main(ActivityThread.java:5083)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:511)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560)
            at dalvik.system.NativeStart.main(Native Method)
     Caused by: java.lang.NullPointerException
            at com.rey.material.widget.ProgressView.getProgressMode(ProgressView.java:125)
            at com.rey.material.widget.ProgressView.onVisibilityChanged(ProgressView.java:98)
            at android.view.View.dispatchVisibilityChanged(View.java:7499)
            at android.view.View.setFlags(View.java:8501)
            at android.view.View.<init>(View.java:3695)
            at com.rey.material.widget.ProgressView.<init>(ProgressView.java:45)
            at com.rey.material.widget.ProgressView.<init>(ProgressView.java:35)
            at java.lang.reflect.Constructor.constructNative(Native Method)
            at java.lang.reflect.Constructor.newInstance(Constructor.java:417)
            at android.view.LayoutInflater.createView(LayoutInflater.java:587)
            at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:687)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:746)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
            at tv.littlemuisc.android.ui.adapter.HomeTestAdapter.onCreateViewHolder(HomeTestAdapter.java:93)
            at android.support.v7.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:4121)
            at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:3431)
            at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:3340)
            at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:1810)
            at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1306)
            at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1269)
            at android.support.v7.widget.LinearLayoutManager.scrollBy(LinearLayoutManager.java:1097)
            at android.support.v7.widget.LinearLayoutManager.scrollVerticallyBy(LinearLayoutManager.java:957)
            at android.support.v7.widget.RecyclerView$ViewFlinger.run(RecyclerView.java:2797)
            at android.view.Choreographer$CallbackRecord.run(Choreographer.java:749)
            at android.view.Choreographer.doCallbacks(Choreographer.java:562)
            at android.view.Choreographer.doFrame(Choreographer.java:531)
            at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:735)
            at android.os.Handler.handleCallback(Handler.java:725)
            at android.os.Handler.dispatchMessage(Handler.java:92)
            at android.os.Looper.loop(Looper.java:137)
            at android.app.ActivityThread.main(ActivityThread.java:5083)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:511)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560)
            at dalvik.system.NativeStart.main(Native Method)

```

**Comment by laylizzlle** [comment]
I don't know why changing the visibility with xml is giving you this error but changing it programmatically works fine for me

**Comment by abiang** [comment]
Thank you, I changing the visibility in the adapter, is works fine.

**Comment by rey5137** [comment]
I have released `1.1.1` version that fixed a dozen bugs in ProgressView. Can you please test it?

**Comment by abiang** [comment]
Yes,I tested the problem has been fixed.
